pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -766,6 +787,10 @@ allChainsLabel:
 			filterChainMatch = chain.FilterChainMatch
 			thriftOpts = configgen.buildSidecarThriftListenerOptsForPortOrUDS(pluginParams)
 
+		case istionetworking.ListenerProtocolDubbo:
+			filterChainMatch = chain.FilterChainMatch
+			dubboOpts = configgen.buildSidecarInboudDubboListenerOptsForPortOrUDS(pluginParams)
```

Comment:
why do you need a separate match? Why can't you move the code into buildInboundNetworkFilters where just like constructing mysql/mongo, etc. you construct dubbo? in fact, thrift should also be moved out like this @howardjohn / @ramaraochavali

Reply:
Dubbo/Thrifit is unlike mysql/mangodb. Dubbo filter handles the routing by itself instead of passing it through to a tcp proxy filter.

The MySQL proxy filter should be chained with the TCP proxy filter as shown in the configuration snippet below:
`filter_chains:
- filters:
  - name: envoy.filters.network.mysql_proxy
    typed_config:
      "@type": type.googleapis.com/envoy.extensions.filters.network.mysql_proxy.v3.MySQLProxy
      stat_prefix: mysql
  - name: envoy.filters.network.tcp_proxy
    typed_config:
      "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
      stat_prefix: tcp
      cluster: ...`

The Dubbo proxy filter has an upstream cluster in its own configuration:

`
filter_chains:
- filters:
  - name: envoy.filters.network.dubbo_proxy
    typed_config:
      "@type": type.googleapis.com/envoy.extensions.filters.network.dubbo_proxy.v3.DubboProxy
      stat_prefix: dubbo_incomming_stats
      protocol_type: Dubbo
      serialization_type: Hessian2
      route_config:
        name: local_route
        interface: org.apache.dubbo.demo.DemoService
        routes:
        - match:
           route:
            cluster: user_service_dubbo_server
`

Reply:
That is perfectly fine. What I am saying is you do not need to worry about building filter chains or expanding this switch case. simply add dubbo as another case here https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/networkfilter.go#L133 .. Look at how redis is done - it is its own filter stack just like dubbo

Reply:
Currently, it would be fine to add dubbo as another case in the buildNetworkFiltersStack function of networkfilter.go because the process for inbound and outbound filter is the same, but they will be different after we add virtual service support as described in phase 2 of the design:https://docs.google.com/document/d/1M2iYOhTf5BZxZXiwAng8dh7XZyu7_0T2AQ1xvl6jlEg/edit?usp=sharing